File: node/models/update-unit-in-army-roster-model.ts
import { db } from '../db';
import { getArmyRosterFromDb } from './get-army-roster-model';

export async function updateUnitInArmyRosterInDb(
    id_in_roster: number,
    isAdding: boolean
) {
    if (id_in_roster === 0) {
        return [];
    }

    const request = new Promise<boolean>((resolve, reject) => {
        db.run(`UPDATE roster SET amount = amount + ${isAdding ? 1 : -1} WHERE id=${id_in_roster};`, function (err) {
            if (err) {
                reject(err);
            }
            return resolve(true);
        });
    });

    try {
        await request;
        return await getArmyRosterFromDb();
    } catch (err) {
        console.error('Error updating unit in Army roster in DB', err);
        return [];
    }
}

File: python/models/update_unit_in_army_roster_model.py
from database import get_db
from models.get_unit_roster_model import get_unit_roster_model


def update_unit_in_roster_model(id_in_roster: int, is_adding: bool):
    try:
        db = get_db()
        db.execute(
            f"""UPDATE roster SET amount = amount + {1 if is_adding else -1} WHERE id={id_in_roster}""")
        print(f"Updated {id_in_roster}")
        db.commit()
        return get_unit_roster_model()
    except Exception as e:
        print("Error posting units to database")
        print(e)
        return []
